### DISCUSSION TITLE
Can you add a changelog

### DISCUSSION THREAD

**Discussion by arandomhooman** [Discord-Oxygen/Discord-Oxygen#22] [Ideas]
Would be very nice @hxr404

**Comment by Anonymous941** [comment]
A changelog for what versions?

**Reply by arandomhooman** [reply]
What features he is developing at the moment, and what he finished.

**Reply by arandomhooman** [reply]
Maybe also tiny snippets of code, or even obfuscated pieces of code that do something, even if it’s glitchy, on the discord website.

**Reply by hxr404** [reply]
> Maybe also tiny snippets of code, or even obfuscated pieces of code that do something, even if it’s glitchy, on the discord website.

That's not going to happen, even if I wanted to provide "snippets" oft my code I couldn't, as it isbthe case with almost every bigger project you can't simply take parts out of a program and expect it to work....
You either run the whole code with all its dependencies on integral core components I've written or you don't run any code at all. Also what would a part of a script which exposes API objects for other modules prove? Even if it worked it wouldn't have any results visible to the user. Having a modular codebase doesn't mean that every single line of code could run on its own. In fact every module that exists and will exists depends on the core components of Discord Oxygen.

**Reply by hxr404** [reply]
> A changelog for what versions?

Yeah that's a really good question.
Discord Oxygen currently is at version dev-5.6.7, assigning version numbers started at dev-0.0.1, all previously written code is unversioned. And before dev-5.* there weren't even proper commits.
And even now my commit messages are written in a way that let's me quickly see what changes are made, but its not clear to others cuz I'm often directly referring to things in the code.
Changelogs like "new class recent chunkapi callback section 3" probably aren't helpful for anyone but me.

**Reply by Anonymous941** [reply]
@hxr404 If you can't share code snippets, can you at least share a screenshot?  Just a screenshot of your script doing something.  Anything.

**Comment by hxr404** [comment]
I'm considering it, thanks for the suggestion. I'm just not really sure whether it's worth the additional work, as I'd spend as much time writing Changogs and Documentation as writing actual code
(For example changes to the API or other core components are really hard to document. And findings of reversed engineering Discord are even harder to explain without leaking the actual code (and rewriting stuff in pseudo-code just to have something documented in my changelog is really not worth the hassle)

**Reply by arandomhooman** [reply]
Maybe only big changes, or full features (or like half or fourth of feature) is complete.
